pkg/endpoint/bpf.go

```diff
@@ -1374,25 +1380,33 @@ func (e *Endpoint) syncPolicyMapWithDump() error {
 	return err
 }
 
+const (
+	// syncPolicyMapInterval is the interval for periodic full reconciliation of
+	// the policy map to catch unexpected discrepancies with agent and kernel state.
+	syncPolicyMapInterval = 15 * time.Minute
+)
+
 func (e *Endpoint) syncPolicyMapController() {
-	ctrlName := fmt.Sprintf("sync-policymap-%d", e.ID)
-	e.controllers.UpdateController(ctrlName,
-		controller.ControllerParams{
-			Group: syncPolicymapControllerGroup,
-			DoFunc: func(ctx context.Context) (reterr error) {
-				// Failure to lock is not an error, it means
-				// that the endpoint was disconnected and we
-				// should exit gracefully.
-				if err := e.lockAlive(); err != nil {
-					return controller.NewExitReason("Endpoint disappeared")
-				}
-				defer e.unlock()
-				return e.syncPolicyMapWithDump()
+	e.startSyncOnce.Do(func() {
+		ctrlName := fmt.Sprintf("sync-policymap-%d", e.ID)
+		e.controllers.UpdateController(ctrlName,
+			controller.ControllerParams{
+				Group: syncPolicymapControllerGroup,
+				DoFunc: func(ctx context.Context) (reterr error) {
```

Comment:
Do we need a named return here?

Reply:
Ah yeah, might as well clean that old code up.